2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,461 @@
+
+I split this proposal into as small milestones as possible. Every milestone ends up with new version of Amy being pushed to production and introduced to all its users. This minimizes the chances of failure and let us adapt or change almost everything after every milestone -- even if it means changing the scope of this project. The milestones are:
+
+1. [Milestone 1: Managing SWC Checkout Process](#milestone-1-managing-swc-checkout-process) -- that is Homework, Discussion Session, Checkout Session and welcoming to the community.
+
+1. [Milestone 2: Managing DC Checkout Process](#milestone-2-managing-dc-checkout-process)
+
+1. [Milestone 3: Managing Recruiting Candidates](#milestone-3-managing-recruiting-candidates) and rest of the workflow, that is matching them to Instructor Trainings and running those trainings.
+
+It may be necessary to move from SQLite to Postgress. I'm discussing in [#716](https://github.com/swcarpentry/amy/issues/716) with Piotr Banaszkiewicz if it's really necessary, but if it is, it must be done at the beginning, as an additional milestone.
+
+I believe it's really important to get feedback as soon as possible, therefore:
+
+- Before introducing new features to all users, they'll be tested by a group of volunteering early users. That way, we don't have to test everything before showing it to first users, so we can get feedback faster and iterate quicker.
+
+- The early users can give feedback in an easy way, e.g. by sending an email rather than opening an issue on github.
+
+- UI mockups will be presented and accepted by the early users.
+
+- Amy dev instance will be available to all interested parties.
+
+It's also important to note what will be **not** implemented:
+
+- Advanced security control. There'll be only one flag in `Person` model that indicates if somebody is a Trainee with very limited permissions or somebody else (Admin, Trainer etc.) who can can edit everything.
+
+- Responsive interface.
+
+- Tracking sessions with Google Analytics or Piwik.
+
+## Milestone 1: Managing SWC Checkout Process
+
+That is everything after an Instructor Training -- that is Homework, Discussion Session, Checkout Session and welcoming to the community.
+
+The new workflow would be:
+
+- At the end of Instructor Training, Trainees register in Amy using the link shared by the Trainer. The link would be unique for each Training, so Trainees can be automatically associated with that Training.
+
+- The Amy interface for new trainees would be completely different from the current one. It could be a single page for the entire Checkout Process, plus one form to update their profile and one to change password.
+
+- Trainees make a change in the SWC Lessons and paste the link to their pull requests into Amy.
+
+- Amy analyzes the link to decide which core lesson did the trainee pick up.
+
+- After entering the pull request, the trainee uses Amy to register for one Discussion Session.
```

Comment:
I don't know yet, but we should design with both approaches in mind - 
one of the signs of a good architecture is that changes like this can be 
made without radically affecting other parts of the program.